CONTRIBUTING.md

```diff
@@ -317,6 +317,8 @@ GitHub provides reactions as a simple and quick way to provide feedback to pull
 
 Pull request reviews should include a list of what has been reviewed in a comment, so other reviewers and mergers can know the state of the review.
 
+Please be considerate about posting reviews that _only_ ask for changes that are stylistic in nature or easily fixed in treewides, so as to avoid stalling the PR.
```

Comment:
```suggestion
Please be considerate about posting review comments that _only_ ask for changes that are stylistic in nature or easily fixed in treewides, so as to avoid stalling the PR.
```


```suggestion
Please be considerate about posting feedback that _only_ ask for changes that are stylistic in nature or easily fixed in treewides, so as to avoid stalling the PR.
```

I think this should relate to review comments or feedback more generally.

Talking about reviews as a whole could imply we're ok with nit-picking when included as part of a larger review.

Taking this further, it could even incentivise artificially inflating the scope of the review, as a means to justify posting less important feedback.

Reply:
Perhaps it should also be specifically talking about comments/feedback that request changes?

I can't speak for others, but personally I'm happy to give or receive feedback that takes the form _"you don't need to change this, but"_.

E.g. _"This isn't important here, but as an FYI the best practice is usually ..."_

Or _"This is purely stylistic, so you're welcome to ignore: if you do ... then nixfmt will handle this better."_

I see these as friendly advice or tips, and I don't think we should demonise this. At least so long as it's made clear they aren't intended as feedback that must be addressed in order to merge a PR.

Reply:
I'm not advocating for a total ban on discussing style, I only want to reduce the number of review cycles. Hence the emphasis on "only".

If a find a mergeable PR where a comment/review asks for a change without explicitly stating they consider it a non-blocker, which happens often, I tend to leave it. 
This is because ignoring such a comment and merging the PR might by some be viewed as a slight, and I'm not the kind of person who likes to risk ruffling someones feathers.

Asking reviewers to explicitly note all their nits as non-blockers is a Sisyphean task, since the alternative is laziness.

> incentivise artificially inflating the scope of the review

If this incentivizes a more thorough review, then I personally consider that a win.
If the reviewer decides to block on some pre-existing issue that is not thematically related to the current PR, I consider that to be [holding the PR hostage](https://www.chiark.greenend.org.uk/~sgtatham/quasiblog/code-review-antipatterns/#ransom-note), which I feel perfectly safe dismissing with a "PR welcome :)".

Reply:
> I'm not advocating for a total ban on discussing style, I only want to reduce the number of review cycles. Hence the emphasis on "only".

I understand and agree with your intent. It's how the current wording could be _interpreted_ that concerns me, now how it was intended.

I think the emphasis should not be put on whether a nitpick is the "only" feedback; rather I think the wording should focus on how "forceful" the nitpick is.

> If the reviewer decides to block on some pre-existing issue that is not thematically related to the current PR, I consider that to be [holding the PR hostage](https://www.chiark.greenend.org.uk/~sgtatham/quasiblog/code-review-antipatterns/#ransom-note), which I feel perfectly safe dismissing with a "PR welcome :)".

IMO focusing on how forceful the nitpick is could also serve to discourage such ransom notes, reducing the number of PRs that need to be dismissed. It also provids a line in CONTRIBUTING that can be referenced when addressing such ransom notes.

Reply:
I agree, and if everyone did as you do during review we would not have the problems we have today. The problem is that the path of least resistance when reviewing is to be as brief as possible. This often means (1) select offending line, (2) click "add suggestion", (3) make changes to offending line, (4) submit. Is that review forceful? Is it a non-blocking nit? It is entirely up to the reader to interpret what the code change suggestion is trying to communicate. If the avatar of the reviewer by chance has a frowny face, are they being forceful? You start grasping at straws.

I fail to see how your two suggestions address forcefulness. Could it be improved?

Reply:
> I fail to see how your two suggestions address forcefulness. Could it be improved?

Correct, my initial suggestions only changed the focus from reviews as a whole to individual review comments.

It was on reflection that I began thinking about how this also relates to how forceful the feedback is.

Perhaps we can iterate on something along these lines?
```suggestion
Please be considerate about posting feedback that _only_ ask for changes that are stylistic in nature or easily fixed in treewides, so as to avoid stalling the PR.
If you insist on posting such feedback, please state clearly that it is a suggestion which should not block merging.
```

Reply:
I think “reviews” here was a deliberate compromise, see e.g. https://github.com/NixOS/nixpkgs/pull/336200#pullrequestreview-2249336189.

While I’m guessing nobody is entirely happy with the current wording, it seems to me to have the useful property that nobody who has commented would strongly object to it, and that is clearly discourages the most pathological review styles while leaving room for differing opinions. I do think “but educating people about conventions is okay if you make it clear that it’s not a hard blocker” wouldn’t go amiss, but if that’s going to stall this further I’d rather avoid the… uh… nitpicking :)

Reply:
The suggestion adds an escape hatch that will IMO only serve to work against the goal reducing review cycles. If a reviewer does post a nit review, what you're then left to argue is its lack of pleasantries to satisfy guidelines, will will in turn spark further arguments.

I'd rather we redirect the nit energy to something more productive, like "express the desired changes in your own PRs.". But adding this to the guideline makes it wordy and unwieldy.